packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,173 @@
+            },
+          },
+        },
+      },
+    });
+
+    expect(parentTemplate.getNestedStack('ChildStack').stack).toMatchTemplate(
+      loadTestFileToJsObject(('child-stack.expected.json')),
+    );
+
+    expect(parentTemplate.getNestedStack('ChildStack').includedTemplate.getNestedStack('GrandChildStack').stack).toMatchTemplate(
+      loadTestFileToJsObject(('grandchild-stack.expected.json')),
+    );
+
+    expect(parentTemplate.getNestedStack('AnotherChildStack').stack).toMatchTemplate(
+      loadTestFileToJsObject(('another-child-stack.expected.json')),
+    );
+
+    expect(parentTemplate.getNestedStack('AnotherChildStack').includedTemplate.getNestedStack('GrandChildStack').stack).toMatchTemplate(
+      loadTestFileToJsObject(('grandchild-stack.expected.json')),
+    );
+  });
+
+  test('throws an error when provided a nested stack that is not present in the template', () => {
+    expect(() => {
+      new inc.CfnInclude(stack, 'ParentStack', {
+        templateFile: testTemplateFilePath('parent-export-stack.json'),
+        nestedStacks: {
+          "FakeStack": {
+            templateFile: testTemplateFilePath('child-import-stack.json'),
+          },
+        },
+      });
+    }).toThrow(/Nested Stack with logical ID 'FakeStack' was not found in the template/);
+  });
+
+  test('Throws an error when NestedStacks contains an ID that is not a CloudFormation::Stack in the template', () => {
+    expect(() => {
+      new inc.CfnInclude(stack, 'ParentStack', {
+        templateFile: testTemplateFilePath('parent-export-stack.json'),
+        nestedStacks: {
+          "ChildStack": {
+            templateFile: testTemplateFilePath('child-import-stack.json'),
+            nestedStacks: {
+              "BucketImport": {
+                templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+              },
+            },
+          },
+        },
+      });
+    }).toThrow(/Nested Stack with logical ID 'BucketImport' is not an AWS::CloudFormation::Stack resource/);
+  });
+
+  test('can modify resources in nested stacks', () => {
+    const parent = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        "ChildStack": {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+          nestedStacks: {
+            "GrandChildStack": {
+              templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+            },
+          },
+        },
+      },
+    });
+
+    const grandChildTemplate = parent.getNestedStack('ChildStack').includedTemplate.getNestedStack('GrandChildStack').includedTemplate;
+    const bucket = grandChildTemplate.getResource('BucketImport') as s3.CfnBucket;
+
+    bucket.bucketName = 'modified-bucket-name';
+
+    const grandChildStack = parent.getNestedStack('ChildStack').includedTemplate.getNestedStack('GrandChildStack').stack;
```

Comment:
Definitely get rid of this repeated and very long `parent.getNestedStack('ChildStack').includedTemplate.getNestedStack('GrandChildStack')` expression by assigning it to an intermediate variable (this is a good general rule: avoid duplicating expressions).